common/src/main/java/com/wynntils/overlays/StatusEffectsOverlay.java

```diff
@@ -108,4 +135,59 @@ private void updateTextRenderSetting() {
     protected TextRenderSetting getTextRenderSetting() {
         return textRenderSetting;
     }
+
+    private Stream<RenderedStatusEffect> stackEffects(List<StatusEffect> effects) {
+        Map<String, RenderedStatusEffect> effectsToRender = new LinkedHashMap<>();
+
+        for (StatusEffect effect : effects) {
+            RenderedStatusEffect entry =
+                    effectsToRender.get(effect.asString().getString());
+            if (entry == null) {
+                entry = new RenderedStatusEffect(effect);
+                effectsToRender.put(effect.asString().getString(), entry);
+            }
+            entry.count += 1;
+        }
+        return effectsToRender.values().stream();
+    }
+
+    private static final class RenderedStatusEffect implements Comparable<RenderedStatusEffect> {
+        private int count = 0;
+        private final StatusEffect effect;
+
+        private RenderedStatusEffect(StyledText name, StyledText modifier, StyledText displayedTime, StyledText prefix) {
+            this.effect = new StatusEffect(name, modifier, displayedTime, prefix);
+        }
+
+        private RenderedStatusEffect(StatusEffect effect) {
+            this.effect = effect;
+        }
+
+        private StyledText getTextToRender() {
+            return this.count > 1
+                    ? StyledText.fromString(this.count + "x ").append(this.effect.asString())
```

Comment:
Before it get's lost, this is yet to be changed. ("2x30% Attack Speed")

Reply:
I can put the "2x" before the modifier (making it something like "<icon> 2x <modifier> <etc>")

Reply:
I'm guessing the "2x" part still needs the same colour as the rest of the message? I tried to do something like that but didn't find how to.

Reply:
```suggestion
                    ? StyledText.fromString(ChatFormatting.GRAY + this.count + "x ").append(this.effect.asString())
```

(or DARK_GRAY, I don't remember)

Reply:
(also, please don't resolve the conversation if the change wasn't done)

Reply:
Alright, I think the conversation it was in had some other things that were added so I resolved it. Will be more careful next time

Reply:
I changed the formatting to grey, I do think the visibility suffers a bit from it.
![stacked_effects2](https://github.com/Wynntils/Artemis/assets/51191570/64eea62f-aca5-474e-85fe-bfe99918b6c0)

Reply:
But it still doesn't use the format I requested. `"2x30% Attack Speed"` is arguably the cleanest way to show this. Can you implement this?

Reply:
It has been done.
![stacked_effects3](https://github.com/Wynntils/Artemis/assets/51191570/cd05d072-d682-46c4-bd65-d5f0ec579ae9)

```java
private StyledText getRenderedText() {
            byte[] modifierBytes = this.effect.getModifier().getString().getBytes(StandardCharsets.UTF_8);
            StyledText modifierText;

            if( modifierBytes.length > 0 && this.count > 1) { // Modifier and stacks
                int start;
                for (start = 0; !Character.isDigit(modifierBytes[start]) && start < modifierBytes.length; start++) {
                }
                start -= 1;
                StyledText modifierStart = this.effect.getModifier().substring(0, start);
                StyledText modifierEnd = this.effect.getModifier().substring(start);
                modifierText = modifierStart
                                .append(StyledText.fromString(ChatFormatting.GRAY + (this.count + "x")))
                                .append(modifierEnd);

            } else if( this.count > 1 ){ // No modifier, does stack
                modifierText = StyledText.fromString(ChatFormatting.GRAY + (this.count + "x"));
            } else { // No stacking
                modifierText = this.effect.getModifier();
            }

            return this.count > 1
                    ? this.effect
                            .getPrefix()
                            .append(StyledText.fromString(" "))
                            .append(modifierText)
                            .append(this.effect.getName())
                            .append(StyledText.fromString(" "))
                            .append(this.effect.getDisplayedTime())
                    : this.effect.
}
```
Not the fanciest java code I've written

Reply:
Oh, god. Wouldn't a simpler way do it?

```
let s = this.effect.getModifier().getString()
let index = 0
if s startsWith - or + then index = 1
s.insert(index, this.count + "x")
```

Reply:
That depends, is there any other prefixes than `+` and `-`?

Reply:
Have you seen any? I have not.

Reply:
Or nothing is a possible prefix.

Reply:
I simplified it a bit, knowing there'll only be a +/-/nothing
```java

        private StyledText getRenderedText() {
            String modifierString = this.effect.getModifier().getString();
            StyledText modifierText;

            int minusIndex = modifierString.indexOf('-');
            int plusIndex = modifierString.indexOf('+');
            int index = max(minusIndex, plusIndex); // look for either a - or a +
            if( index < 0 ){// We can simply put the count string at the start
                modifierText = (this.count > 1 ? StyledText.fromString(ChatFormatting.GRAY + (this.count + "x")) : StyledText.EMPTY)
                        .append(this.effect.getModifier());
            } else { // The count string is inserted between the +/- and the number
                index += 1;
                modifierText = StyledText.fromString(modifierString.substring(0, index))
                        .append(this.count > 1 ? StyledText.fromString(ChatFormatting.GRAY + (this.count + "x")) : StyledText.EMPTY)
                        .append(StyledText.fromString(ChatFormatting.GRAY + modifierString.substring(index)));
            }

            return this.count > 1
                    ? this.effect
                            .getPrefix()
                            .append(StyledText.fromString(" "))
                            .append(modifierText)
                            .append(this.effect.getName())
                            .append(StyledText.fromString(" "))
                            .append(this.effect.getDisplayedTime())
                    : this.effect.asString();
        }
```

Reply:
Can you push this so I can comment on it properly?

1, You should return early if the count is 1.
2, index < 0 should be index == -1
3, move the comments to be above the lines, not "next" to them
4, you do a lot of StyledText.fromString and append calls. Use string concatenation and call StyledText.fromString once. It will make the code cleaner (talking about the if-else calls).

Reply:
It was pushed when I wrote the message.

I've processed all of these points in the newly pushed code now.